File: LeetCode/June LeetCoding Challenge/June9_IsSubsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.size() > t.size())
            return false;
        if(s.size() == 0 || t.size()==0)
            return true;
        char c = s[0];
        int j=0;
        for(int i=0;i<t.size();i++)
        {
            if(c==t[i])
            {
                j++;
                if(j== s.size())
                    return true;
                else
                    c=s[j];
            }

        }
        return false;
    }
};

File: LeetCode/June LeetCoding Challenge/June9_IsSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s)>len(t):
            return False
        if(len(s)==0 or len(t)==0):
            return True
        c = s[0]
        j=0
        for i in range(len(t)):
            if c==t[i]:
                j+=1
                if j==len(s):
                    return True
                else:
                    c=s[j]
        return False
        
